### PR TITLE
CI: improve test:multiverse:clobber

### LINKED ISSUE #964: rake multiverse:clobber - needs Postgres support and needs better rescuing
## Description
The multiverse tests offer `rake multiverse clobber` for cleanup. There are two problems with that task:

* Postgres tests have been added, but the task still only drops MySQL databases.
* Exceptions thrown by the current MySQL database drop operations should be rescued individually, but currently they are erroneously caught with a single `rescue` that states that there was a problem with the listing of the databases instead of with the dropping of them (see `remove_local_multiverse_databases` in `lib/tasks/multiverse.rb`

## Expected Behavior
* Postgres databases should be dropped by the clobber task
* Exceptions related to the dropping of a database should not be incorrectly reported as issues with listing databases.

## [Troubleshooting](https://discuss.newrelic.com/t/troubleshooting-frameworks/108787) or [NR Diag](https://docs.newrelic.com/docs/using-new-relic/cross-product-functions/troubleshooting/new-relic-diagnostics) results
Provide any other relevant log data.
TIP:  Scrub logs and diagnostic information for sensitive information 
n/a

## Steps to Reproduce
Please be as specific as possible.
TIP:  Link a sample application that demonstrates the issue. 
* Run the task after running a Postgres suite that creates (a) Postgres database(s)
* Fake an exception with MySQL database dropping

## Your Environment
Include as many relevant details about your environment as possible including the running version of the Ruby agent, the Ruby version being used and any other relevant environment information
(all)

## Additional context
Add any other context about the problem here. For example, relevant community posts or support tickets.
n/a

### PR DESCRIPTION
**Pull request by fallwith**
`rake test:multiverse:clobber` improvements

- refactor the `Removers` module into a new `Multiverse::Clobber` class, and have it live beneath `test/multiverse`
- add rake content that is broken without `test/` content to `.build_ignore`
- update `rake test:multiverse:clobber` to delete Postgresql databases in addition to MySQL databases. Errors encountered for one will not prevent attempts to delete databases for the other.
- update `rake test:multiverse:clobber` with individual rescues for each operation; especially individual database drop calls
- add additional `puts` statements for improved output

resolves #964

### CODE REVIEW

File: test/multiverse/lib/multiverse/clobber.rb [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+module Multiverse
+  class Clobber
+    EXECUTABLES = {mysql: %w[mysql],
+                   postgresql: %w[awk dropdb psql]}
+    LIST_COMMANDS = {mysql: %(echo "show databases" | mysql -u root),
+                     postgresql: %(echo '\\l' |psql -d postgres|awk '{print $1}')}
+    PARTIALDATABASE_NAME = 'multiverse'
```
Comment (kaylareopelle) [inline test/multiverse/lib/multiverse/clobber.rb:11]:
```suggestion
    PARTIAL_DATABASE_NAME = 'multiverse'
```

File: test/multiverse/lib/multiverse/clobber.rb [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      raise "Unable to locate the following executables in your PATH: #{missing}"
+    end
+
+    def remove_databases(db_type)
+      databases_list(db_type).each { |database| drop_database(db_type, database) }
+    end
+
+    def databases_list(db_type)
+      puts "Obtaining a list of #{db_type} databases..."
+      `#{LIST_COMMANDS[db_type]}`.chomp!.split("\n").select { |s| s.include?(PARTIALDATABASE_NAME) }
```
Comment (kaylareopelle) [inline test/multiverse/lib/multiverse/clobber.rb:63]:
```suggestion
      `#{LIST_COMMANDS[db_type]}`.chomp!.split("\n").select { |s| s.include?(PARTIAL_DATABASE_NAME) }
```

File: .build_ignore [hunk 3]
```diff
@@ -19,3 +19,8 @@ lefthook.yml
 log/
 README.md
 test/
+lib/tasks/bump_version.rb
+lib/tasks/coverage_report.rb
+lib/tasks/mutliverse.rake
+lib/tasks/mutliverse.rb
```
Comment (kaylareopelle) [inline .build_ignore:25]:
```suggestion
lib/tasks/multiverse.rake
lib/tasks/multiverse.rb
```

**Review by kaylareopelle — COMMENTED** [review]
These changes are great. My repo feels so much cleaner. 🧹  

One small thing -- on my machine, I needed to use `rake test:multiverse:clobber` to get the task to run. Were you able to use only `rake multiverse:clobber`?

**Comment by fallwith** [comment]
> These changes are great. My repo feels so much cleaner. 🧹
> 
> One small thing -- on my machine, I needed to use `rake test:multiverse:clobber` to get the task to run. Were you able to use only `rake multiverse:clobber`?

Ah, whoops. Thanks, @kaylareopelle. The task is indeed `test:multiverse:clobber`. That part hasn't been changed, I just typed it wrong in the PR title and description. The title and description have been updated.

**Review by tannalynn — APPROVED** [review]
This is great! TY

**Pull request merged**